File: app/src/main/java/awais/instagrabber/repositories/responses/directmessages/DirectInboxResponse.java
package awais.instagrabber.repositories.responses.directmessages;

import awais.instagrabber.repositories.responses.User;

public class DirectInboxResponse {
    private final User viewer;
    private final DirectInbox inbox;
    private final long seqId;
    private final long snapshotAtMs;
    private final int pendingRequestsTotal;
    private final User mostRecentInviter;
    private final String status;

    public DirectInboxResponse(final User viewer,
                               final DirectInbox inbox,
                               final long seqId,
                               final long snapshotAtMs,
                               final int pendingRequestsTotal,
                               final User mostRecentInviter,
                               final String status) {
        this.viewer = viewer;
        this.inbox = inbox;
        this.seqId = seqId;
        this.snapshotAtMs = snapshotAtMs;
        this.pendingRequestsTotal = pendingRequestsTotal;
        this.mostRecentInviter = mostRecentInviter;
        this.status = status;
    }

    public User getViewer() {
        return viewer;
    }

    public DirectInbox getInbox() {
        return inbox;
    }

    public long getSeqId() {
        return seqId;
    }

    public long getSnapshotAtMs() {
        return snapshotAtMs;
    }

    public int getPendingRequestsTotal() {
        return pendingRequestsTotal;
    }

    public User getMostRecentInviter() {
        return mostRecentInviter;
    }

    public String getStatus() {
        return status;
    }
}

File: app/src/main/java/awais/instagrabber/repositories/responses/directmessages/DirectInboxResponse.kt
package awais.instagrabber.repositories.responses.directmessages

import awais.instagrabber.repositories.responses.User

data class DirectInboxResponse(
    val viewer: User? = null,
    val inbox: DirectInbox? = null,
    val seqId: Long = 0,
    val snapshotAtMs: Long = 0,
    val pendingRequestsTotal: Int = 0,
    val hasPendingTopRequests: Boolean = false,
    val mostRecentInviter: User? = null,
    val status: String? = null,
)